### PR TITLE
feat(mentions): mentions + notifications

### PR DESCRIPTION
**Pull request by bautistaaa**
### Description
Enables the ability to tag any user in comments.
Enables Notifications. See below for behavior

known issue: 
- you can duplicate notifications (e.g: unlike and like a comment will send another notification)
- editing a comment and tagging someone will not tag them
- unliking/deleting a reply will not remove the notification
- no loader styling

should be notified when
behavior:
- someone likes my comment
- mentions me in a comment
- replies to my comment
- likes my shared solution
- comments on my shared solution

closes https://github.com/typehero/typehero/issues/1478
closes https://github.com/typehero/typehero/issues/1477

### CODE REVIEW

File: apps/web/src/components/mention/actions.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+'use server';
+
+import { prisma } from '@repo/db';
+
+export async function searchUsers(query: string) {
+  return prisma.user.findMany({
+    where: {
+      name: {
+        contains: query,
```
Comment (Hacksore) [inline apps/web/src/components/mention/actions.ts:9]: limit this to the max username length to prevent abuse.

Comment (bautistaaa) [reply apps/web/src/components/mention/actions.ts:9]: FIIIIINE

**Comment by Hacksore** [comment]
looking good!

image

Only thing that is a bit annoying is github doesnt use md so you can see if I just do `@bautistaaa` (@bautistaaa) i don't need to actually interact with the autocomplete menu.

See the middle in the image above one where I just typed it 😂.

Probably it is what it is but how hard it it be to just make @ mentions work without markdown?

**Comment by bautistaaa** [comment]
> looking good!
> 
> image
> Only thing that is a bit annoying is github doesnt use md so you can see if I just do `@bautistaaa` (@bautistaaa) i don't need to actually interact with the autocomplete menu.
> 
> See the middle in the image above one where I just typed it 😂.
> 
> Probably it is what it is but how hard it it be to just make @ mentions work without markdown?

I think I'd need some custom plugin to pick up `@` and convert to an anchor tag. I'll check

**Comment by Hacksore** [comment]
@bautistaaa yeah you can write a custom rehype/remark plugin to do this. 

Here we have one here as an example:
https://github.com/typehero/typehero/blob/ef3a5c1242b96374c15d6284cb82577fb3c14b0b/packages/ui/src/components/markdown.tsx#L23

**Comment by Hacksore** [comment]
when can i test this @bautistaaa 😂

**Comment by Hacksore** [comment]
feedback so far

Fix the overflow
https://typehero-mul19j2op-trash-company.vercel.app/challenge/generic-function-arguments/comments/8?replyId=11

Fix the caching issue

**Comment by Hacksore** [comment]
perhaps limit to 99+?

image

**Comment by Hacksore** [comment]
minor inconsistenty is the title does not have a suffix like other pages do.
image

ex:
image

**Comment by Hacksore** [comment]
suggesting to have some "not found" state rather then just super dead.

image

**Comment by Hacksore** [comment]
i was able to noti myself on `SeanAWS` with this payload
```
testing @seanaws
```

do we account for case?

**Pull request merged**